subsys/bluetooth/host/hci_core.c

```diff
@@ -511,6 +511,18 @@ static void set_phy_conn_param(const struct bt_conn *conn,
 	phy->max_ce_len = 0;
 }
 
+/** Convert the argument to human readable string.
+ *  The resulting string has lifetime until end of the enclosing block.
+ */
+#define STR(x) _Generic((x), \
+	const bt_addr_le_t * : STR_ADDR(x))
+#define STR_ADDR(addr) str_addr(addr, (char[BT_ADDR_LE_STR_LEN]) {})
+static char *str_addr(const bt_addr_le_t *addr, char *str)
+{
+	bt_addr_le_to_str(addr, str, BT_ADDR_LE_STR_LEN);
+	return str;
+}
```

Comment:
This basically looks like a copy of `bt_addr_str/bt_addr_str_real`:

```c
const char *bt_addr_str_real(const bt_addr_t *addr)
{
	static char str[BT_ADDR_STR_LEN];

	bt_addr_to_str(addr, str, sizeof(str));

	return str;
}
```

Reply:
Good point. I did not know it existed. Mine is potentially a bit better for DBG because it allows printing two addresses in one line. (The static variable in `bt_addr_str_real` makes that awkward.)

Reply:
Will possibly need to verify if that's still the case after switching to the new version of the logging system. May simply have been missing strdup

Reply:
It looks like `log_strdup` is no longer needed after Logging v1 was removed in v2.6. `log_strdup` was removed in 041f0e5379c660d73c0fe69322b8642470316a22.

I will update this PR to use `bt_addr_str`. It's clear that `BT_DBG` and `bt_addr_str` are meant to be used together since they are defined in the same file `bluetooth/common/log.h`.

I'll create a new PR with my thread-safe implementation of `bt_addr_str`.